### ISSUE TITLE
Peer group support in vrf context

### ISSUE THREAD

**Issue by daanvdsanden** [aristanetworks/ansible-avd#874]
**Issue Type**

- [x] Template enhancement
- [x] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**
I would like to use peer groups for BGP peerings in a VRF. 

**Describe the solution you'd like**
An update to the tenant vrf model in `eos_designs`:
```yaml
tenants:
  < tenant_a >:
    vrfs:
      < tenant_a_vrf_1 >:
        bgp_peer_groups:
          < peer_group_name >:
            remote_as: < remote ASN >
            description: < description >
            password: < encrypted password >
            send_community: < standard | extended | large | all >
            next_hop_self: < true | false >
            maximum_routes: < 0-4294967294 >
            default_originate:
              always: < true | false >
            update_source: < interface >
            bfd: < true | false >
            address_families:
              <address_family>:
                activate: < true | false >
            ebgp_multihop: < 1-255 >
            # Nodes is required to restrict configuration of BGP peer groups to certain nodes in the network.
            nodes: [ < node_1 >, < node_2> ]
            # Next hop settings can be either ipv4 or ipv6 for one neighbor, this will be applied by a uniquely generated route-map per neighbor.
            # Next hop takes precedence over route_map_out.
            set_ipv4_next_hop: < IPv4_address >
            set_ipv6_next_hop: < IPv6_address >
            route_map_out: < route-map name >
            route_map_in: < route-map name >
            local_as: < local BGP ASN >
        bgp_peers:
          < neighbor >:
            peer_group: < peer_group_name >
```

**Describe alternatives you've considered**

At the moment I'm using per neighbor configuration. But updating multiple sessions that must have the same settings takes a long time and is error prone. This also makes the VRF tenant yaml more readable.

**Additional context**

~~Add any other context or screenshots about the feature request here.~~

**Comment by carlbuchmann** [comment]
@daanvdsanden this is a great suggestion. As a workaround, you should be able to add the peer group leveraging [custom EOS structured configuration](https://avd.sh/en/latest/roles/eos_designs/doc/l3ls-evpn/common-settings/#custom-eos-structured-configuration) and then reference it in `bgp_peers`.

**Comment by daanvdsanden** [comment]
@carlbuchmann thanks for the suggestion.

I do like the config to be in the Tenant structure, because when the tenant is removed all the config with it will also go away/be cleaned up.

I do have some working templates, which I'm more than willing to share, given the proposed structure is ok. But I need some help or guidance in writing the correct tests with it.

**Comment by kmueller68** [comment]
@carlbuchmann do you mean, that we should already now be able to reference an existing peer-group in tenant vrf like this?
bgp_peers:
          < neighbor >:
            peer_group: < peer_group_name >

Thank you, Karl

**Comment by daanvdsanden** [comment]
> @carlbuchmann do you mean, that we should already now be able to reference an existing peer-group in tenant vrf like this?
> bgp_peers:
> < neighbor >:
> peer_group: < peer_group_name >

Yes you can already reference to an existing peer-group.

**Comment by carlbuchmann** [comment]
@daanvdsanden - thank you for confirming! I will update the documentation accordingly. Under the hood the [tenant EVPN template](https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_designs/templates/designs/l3ls-evpn/tenant_evpn_vxlan/leaf-tenant-router-bgp-evpn-vrfs.j2) passes all input vars for `bgp_peers` to eos_cli_config_gen but we chose to only document what was tested. Note that this behavior may change in the future, and we would only pass vars that are documented.

@daanvdsanden - yes I agree that it very practical to also define the peer group in the Tenant construct. We are in the process of refactoring eos_designs, so this is why I have assigned this issue to milestone 2.2.0, to avoid merge conflicts. Once 2.1.0 is released, I would be happy to help you with this PR and show you how to add molecule test cases.
